app/lib/actions/katello/check_matching_content.rb

```diff
@@ -0,0 +1,17 @@
+module Actions
+  module Katello
+    module CheckMatchingContent
+      def check_matching_content(new_repository, source_repositories)
+        check_matching_content = ::Katello::RepositoryTypeManager.find(new_repository.content_type).metadata_publish_matching_check
+        if new_repository.environment && source_repositories.count == 1 && check_matching_content
```

Comment:
I'm finding that `matching_content` is false for my concurrent `IndexContent` still because `new_repository.environment` is `nil`.  It seems like this is always the case for me.

Reply:
Great catch!

Reply:
As per discussion with @jlsherrill if the target repo is an archive repo, this is expected.